File: 2026POTD/JANUARY/15-01-2026/2943.cs
using System;

public class Solution {
    public int MaximizeSquareHoleArea(int n, int m, int[] hBars, int[] vBars) {
        int h = Longest(hBars);
        int v = Longest(vBars);
        int side = Math.Min(h, v);
        return side * side;
    }

    private int Longest(int[] a) {
        Array.Sort(a);
        int best = 1, cur = 1;
        for (int i = 1; i < a.Length; i++) {
            if (a[i] == a[i - 1] + 1) cur++;
            else cur = 1;
            if (cur > best) best = cur;
        }
        return best + 1;
    }
}

File: 2026POTD/JANUARY/15-01-2026/2943.py
class Solution(object):
    def maximizeSquareHoleArea(self, n, m, hBars, vBars):
        def longest(a):
            a.sort()
            best = cur = 1
            for i in range(1, len(a)):
                if a[i] == a[i - 1] + 1:
                    cur += 1
                else:
                    cur = 1
                if cur > best:
                    best = cur
            return best + 1

        side = min(longest(hBars), longest(vBars))
        return side * side